### PR TITLE
added logic to pass config data settings to rego rules

### PR DESCRIPTION
**Pull request by AkhtarAmir**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -76,3 +76,7 @@ outdated-api-updated:
 .PHONY: bundle
 bundle:
 	./scripts/bundle.sh
+
+.PHONY: build
+build:
+	go build -o bin/defsec ./cmd/defsec
```
Comment (AkhtarAmir) [inline Makefile:82]:
```suggestion
```

File: pkg/scanners/cloud/aws/options.go [hunk 2]
```diff
@@ -31,6 +31,14 @@ func ScannerWithAWSRegion(region string) options.ScannerOption {
...
 }
 
+func ScannerWithRegoOnly(regoOnly bool) options.ScannerOption {
+	return func(s options.ConfigurableScanner) {
+		if aws, ok := s.(ConfigurableAWSScanner); ok {
+			aws.SetRegoOnly(regoOnly)
+		}
+	}
+}
+
```
Comment (AkhtarAmir) [inline pkg/scanners/cloud/aws/options.go:41]:
```suggestion
```

**Comment by simar7** [comment]
Could you add some unit tests that exercise this logic? You can see some here for example on how to write one https://github.com/aquasecurity/defsec/blob/master/pkg/scanners/cloud/aws/scanner_test.go

File: pkg/scanners/cloud/aws/scanner_test.go [hunk 3]
```diff
@@ -62,6 +62,7 @@ func Test_AWSInputSelectors(t *testing.T) {
 	testCases := []struct {
 		name            string
 		srcFS           fs.FS
+		dataFS          fs.FS
```
Comment (simar7) [inline pkg/scanners/cloud/aws/scanner_test.go:65]: I see you've added this field but haven't defined any cases in this test. I think we should move the test you added as a test case here as most of the logic is common between the two.

File: pkg/scanners/cloud/aws/scanner_test.go [hunk 4]
```diff
@@ -437,3 +439,98 @@ deny[res] {
...
+		srcFS           fs.FS
+		dataFS          fs.FS
+		state           state.State
+		expectedResults struct {
+			totalResults int
+			summaries    []string
+		}
+	}{
+		{
+			name: "single cloud, single selector with config data",
```
Comment (simar7) [inline pkg/scanners/cloud/aws/scanner_test.go:455]: This is good. We can add another case where the value is a different type, for instance an int.

**Review by simar7 — CHANGES_REQUESTED** [review]
Good start, left some comments.

File: pkg/scanners/cloud/aws/scanner_test.go (hunk 3 above)
Comment (AkhtarAmir) [reply pkg/scanners/cloud/aws/scanner_test.go:65]: We are adding to these options in scanner
options.ScannerWithDataFilesystem(tc.dataFS),
options.ScannerWithDataDirs("config-data/")

so we will have to include config data this in all tests

File: pkg/scanners/cloud/aws/scanner_test.go (hunk 4 above)
Comment (AkhtarAmir) [reply pkg/scanners/cloud/aws/scanner_test.go:455]: okay

File: pkg/scanners/cloud/aws/scanner_test.go (hunk 3 above)
Comment (simar7) [reply pkg/scanners/cloud/aws/scanner_test.go:65]: You can make the options on the test cases itself rather than declared once.

Comment (simar7) [reply pkg/scanners/cloud/aws/scanner_test.go:65]: Any update?

**Comment by simar7** [comment]
@AkhtarAmir can you update this PR soon? I'm looking to cut a release end of the month (regular cadence for trivy and defsec) and would like to have this included.

**Pull request closed without merging**